### PR TITLE
[modules] Swarming

### PR DESCRIPTION
**Pull request by kirkscheper**
Some code I have been working on for a while. It implements swarming code using potential fields or a neural network. It uses the new traffic info code and the new utm helpers so may be broken till those changes are integrated.

I also used this to add my personal folder to pprz.

### CODE REVIEW

File: conf/modules/extra_dl.xml [hunk 1]
```diff
@@ -3,21 +3,32 @@
...
   </header>
   <init fun="extra_pprz_dl_init()"/>
   <event fun="ExtraDatalinkEvent()"/>
   <makefile target="ap">
-    <configure name="EXTRA_DL_PORT" case="upper|lower"/>
+    <configure name="EXTRA_DL_PORT" default="uart1" case="upper|lower"/>
+    <raw>
+      ifneq (,$(findstring udp,$(EXTRA_DL_PORT_LOWER)))
+      include $(CFG_SHARED)/udp.makefile
+      $(TARGET).CFLAGS += -DEXTRA_DOWNLINK_DEVICE=$(EXTRA_DL_PORT_LOWER)
```
Comment (flixr) [inline conf/modules/extra_dl.xml]: this adds the flags twice...

File: sw/airborne/subsystems/ins/vf_extended_float.c [hunk 2]
```diff
@@ -351,3 +351,55 @@ void vff_realign(float z_meas)
...
+  // update covariance
+  Pp = Pm - K*H*Pm;
+*/
+static inline void update_vz_conf(float vz, float conf)
+{
+  const float yd = vz - vff.zdot;
+  const float S = vff.P[1][1] + conf;
+  const float K1 = vff.P[0][1] * 1 / S;
+  const float K2 = vff.P[1][1] * 1 / S;
+  const float K3 = vff.P[2][1] * 1 / S;
```
Comment (flixr) [inline sw/airborne/subsystems/ins/vf_extended_float.c]: This is not correct, vf_extended has one state more that has to be taken into account here...

**Comment by kirkscheper** [comment]
Oew, over looked that. tnx

File: conf/modules/swarm_nn.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="swarm_nn" dir="multi">
+  <doc>
+    <description>
+      Potential fields swarm flocking.
```
Comment (flixr) [inline conf/modules/swarm_nn.xml]: wrong brief description, some more details would also be nice ;-)

**Comment by flixr** [comment]
Added the extra_dl fix with 5626b5114707e122b4a9d6b21a4322eb466ff915 and vertical speed update for vf_extended with 22cf2e680969aef3c97d6a2cb87bcbb44fc5b87e

**Comment by flixr** [comment]
The swarming modules will have to wait for the traffic_info from #1638

**Comment by noether** [comment]
Hi @kirkscheper , is it possible to check somewhere the documentation/papers that you were using for your implementation?

**Comment by kirkscheper** [comment]
I haven't written anything yet. When we have settled on the final version I will write/update a wiki page for this.

**Comment by podhrmic** [comment]
@kirkscheper any updates on this?

**Comment by kirkscheper** [comment]
@podhrmic not really, I actually think I will not try to reintegrate this anymore, it is difficult to introduce a Neural network to PPRZ without making a more generic framework.

**Pull request closed without merging**